src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2958,6 +2962,18 @@ public void visitNewClass(JCNewClass tree) {
             } else {
                 // nested class
                 thisArg = makeOwnerThis(tree.pos(), c, false);
+                if (currentMethodSym != null &&
+                        ((currentMethodSym.flags_field & (STATIC | BLOCK)) == BLOCK) &&
+                        currentMethodSym.owner.isValueClass()) {
+                    // instance initializer in a value class
+                    Set<JCExpression> outerThisSet = initializerOuterThis.get(currentClass);
+                    if (outerThisSet == null) {
+                        outerThisSet = Set.of(thisArg);
+                    } else {
+                        outerThisSet.add(thisArg);
```

Comment:
Is this `java.util.Set`? If yes then `add` probably doesn't work on immutable sets.

Reply:
got it, good catch, will fix thanks!